pioinstaller/penv.py

```
@@ -402,6 +402,39 @@ def install_uv_in_venv_with_system_uv(system_uv_exe, penv_dir):
 
+def install_pip_in_venv_with_penv_uv(penv_dir):
+    """
+    Use the penv's own uv to install pip into the penv.
+    Required for backwards compatibility with old PlatformIO versions.
+    """
+    penv_uv = os.path.join(get_penv_bin_dir(penv_dir), UV_EXE)
+    venv_python = os.path.join(get_penv_bin_dir(penv_dir), PYTHON_EXE)
+    cmd = [penv_uv, "pip", "install", "--python", venv_python, "pip"]
+
+    try:
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)